utils/swift_build_support/swift_build_support/products/curl.py

```diff
@@ -0,0 +1,136 @@
+        """product_source_name -> Bool
+
+        returns 'curl'
+        """
+        return "curl"
+
+    @classmethod
+    def get_dependencies(cls):
+        return [zlib.Zlib]
+
+    def should_build(self, host_target):
+        """should_build() -> Bool
+
+        Return True if curl should be built
+        """
+        return self.args.build_curl
+
+    def should_test(self, host_target):
+        """should_test() -> Bool
+
+        Returns True if curl should be tested.
+        Currently is set to false
+        """
+        return False
+
+    def should_install(self, host_target):
+        """should_install() -> Bool
+
+        Returns True
+        If we're building curl, you're going to need it
+        """
+        return self.args.build_curl
+
+    def install(self, host_target):
+        """
+        Install curl to the target location
+        """
+        path = self.host_install_destdir(host_target)
+        self.install_with_cmake(['install'], path)
+
+    def build(self, host_target):
+        if self.args.build_static_curl:
+            self.cmake_options.define('BUILD_SHARED_LIBS', 'NO')
+            self.cmake_options.define('CMAKE_POSITION_INDEPENDENT_CODE', 'YES')
+        else:
+            self.cmake_options.define('BUILD_SHARED_LIBS', 'YES')
+
+        if self.args.curl_build_variant is None:
+            self.args.curl_build_variant = 'Release'
+        self.cmake_options.define('CMAKE_BUILD_TYPE:STRING',
+                                  self.args.curl_build_variant)
+        self.cmake_options.define('CMAKE_INSTALL_LIBDIR', 'lib')
+        self.cmake_options.define('CMAKE_INSTALL_PREFIX', '/usr')
+        self.cmake_options.define('BUILD_CURL_EXE', 'NO')
+        self.cmake_options.define('CMAKE_USE_OPENSSL', 'NO')
+        self.cmake_options.define('CURL_CA_PATH', 'none')
+        self.cmake_options.define('CMAKE_USE_SCHANNEL', 'NO')
+        self.cmake_options.define('CMAKE_USE_LIBSSH2', 'NO')
+        self.cmake_options.define('HAVE_POLL_FINE', 'NO')
+        self.cmake_options.define('CURL_DISABLE_LDAP', 'YES')
+        self.cmake_options.define('CURL_DISABLE_LDAPS', 'YES')
+        self.cmake_options.define('CURL_DISABLE_TELNET', 'YES')
+        self.cmake_options.define('CURL_DISABLE_DICT', 'YES')
+        self.cmake_options.define('CURL_DISABLE_FILE', 'YES')
+        self.cmake_options.define('CURL_DISABLE_TFTP', 'YES')
+        self.cmake_options.define('CURL_DISABLE_RTSP', 'YES')
+        self.cmake_options.define('CURL_DISABLE_PROXY', 'YES')
+        self.cmake_options.define('CURL_DISABLE_POP3', 'YES')
+        self.cmake_options.define('CURL_DISABLE_IMAP', 'YES')
+        self.cmake_options.define('CURL_DISABLE_SMTP', 'YES')
+        self.cmake_options.define('CURL_DISABLE_GOPHER', 'YES')
+        self.cmake_options.define('CURL_ZLIB', 'YES')
+        self.cmake_options.define('ENABLE_UNIX_SOCKETS', 'NO')
+        self.cmake_options.define('ENABLE_THREADED_RESOLVER', 'NO')
```

Comment:
Does this match the windows configuration?

Reply:
If I flip `SCHANNEL` to `YES`, it should match https://github.com/apple/swift/blob/64e9d0b27f3b8f33a8b7c8533b215562bc97349b/utils/build-windows-toolchain.bat#L133-L153

Reply:
Yes, cannot enable that on non-Windows, that makes sense.